decompiler/backend/codevisitor.py

```diff
@@ -70,10 +71,30 @@ def visit_condition_node(self, node: ast_nodes.ConditionNode) -> str:
         true_str = self.visit(node.true_branch_child)
         if node.false_branch is None:
             return f"if ({self._condition_string(node.condition)}) {{{true_str}}}"
+
         false_str = self.visit(node.false_branch_child)
-        if isinstance(node.false_branch_child, ast_nodes.ConditionNode):
-            return f"if ({self._condition_string(node.condition)}){{{true_str}}} else {false_str}"
-        return f"if ({self._condition_string(node.condition)}){{{true_str}}} else{{{false_str}}}"
+
+        condition = node.condition
+        true_child = node.true_branch_child
+        false_child = node.false_branch_child
+
+        def swap_branches():
+            nonlocal condition, true_child, false_child, true_str, false_str
+            condition = ~condition
+            true_child, false_child = false_child, true_child
+            true_str, false_str = false_str, true_str
+
+        length_comparisons = {"none": False, "smallest": len(true_str) > len(false_str), "largest": len(true_str) < len(false_str)}
+        if length_comparisons[self._preferred_true_branch]:
+            swap_branches()
+
+        if isinstance(true_child, ast_nodes.ConditionNode) or isinstance(false_child, ast_nodes.ConditionNode):
+            if not isinstance(false_child, ast_nodes.ConditionNode):
+                swap_branches()
+
+            return f"if ({self._condition_string(condition)}) {{{true_str}}} else {false_str}"
+        else:
+            return f"if ({self._condition_string(node.condition)}) {{{true_str}}} else {{{false_str}}}"
```

Comment:
Couldn't you simplify it to the following?
```suggestion
        if isinstance(true_child, ast_nodes.ConditionNode) and not isinstance(false_child, ast_nodes.ConditionNode):
            swap_branches()

        return f"if ({self._condition_string(condition)}) {{{true_str}}} else {false_str}"
```

Reply:
The difference between the two branches is that one branch print the if-else without brackets for the else branch and the other one does. This is done so that else if chaining doesn't look like ... else { if () .. } but like ... else if () ...